recipes/poco/all/conanfile.py

```diff
@@ -112,14 +112,58 @@ def configure(self):
 
     def requirements(self):
+        self.requires("pcre/8.41")
+        self.requires("zlib/1.2.11")
+        if self.options.enable_xml:
+            self.requires("expat/2.2.9")
+        if self.options.enable_data_sqlite:
+            self.requires("sqlite3/3.31.1")
+        if self.options.enable_apacheconnector:
+            self.requires("apr/1.7.0")
+            self.requires("apr-util/1.6.1")
+            raise ConanInvalidConfiguration("apache2 is not (yet) available on CCI")
+            self.requires("apache2/x.y.z")
         if self.options.enable_netssl or \
-           self.options.enable_netssl_win or \
            self.options.enable_crypto or \
            self.options.force_openssl or \
            self.options.get_safe("enable_jwt", False):
             self.requires("openssl/1.1.1g")
 
-    def _patch(self):
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        if Version(self.version) < "1.10":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "Data", "SQLite", "CMakeLists.txt"),
+                                  "find_package(SQLite3)", "find_package(sqlite3 REQUIRED)") #1.8.1
+            sqlite3_replace = "${SQLITE3_LIBRARIES}"
+            tools.replace_in_file(os.path.join(self._source_subfolder, "Data", "SQLite", "CMakeLists.txt"),
+                                  "SQLITE3_INCLUDE_DIRS", "sqlite3_INCLUDE_DIRS")
+        else:
+            tools.replace_in_file(os.path.join(self._source_subfolder, "Data", "SQLite", "CMakeLists.txt"),
+                                  "find_package(SQLite3 REQUIRED)", "find_package(sqlite3 REQUIRED)") #1.8.1
+            sqlite3_replace = "SQLite::SQLite3"
+        tools.replace_in_file(os.path.join(self._source_subfolder, "Data", "SQLite", "CMakeLists.txt"),
+                              sqlite3_replace, "sqlite3::sqlite3")
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "Foundation", "CMakeLists.txt"),
+                              "find_package(PCRE REQUIRED)", "find_package(pcre REQUIRED)")
+        pcre_replace = "${PCRE_LIBRARIES}"
+        if Version(self.version) >= "1.10":
+            pcre_replace = "Pcre::Pcre"
+        tools.replace_in_file(os.path.join(self._source_subfolder, "Foundation", "CMakeLists.txt"),
+                              pcre_replace, "pcre::pcre")
+
+        if Version(self.version) < "1.10":
+            openssl_replace = "${OPENSSL_LIBRARIES}"
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                                  "OPENSSL_INCLUDE_DIR", "OpenSSL_INCLUDE_DIR")
+        else:
+            # FIXME: conan components should make the following line unnecessary
+            openssl_replace = "OpenSSL::SSL OpenSSL::Crypto"
+        tools.replace_in_file(os.path.join(self._source_subfolder, "Crypto", "CMakeLists.txt"),
```

Comment:
Hello @madebr 
Just a question, I havent look entirely in depth for this.
but isnt the current cci recipe for poco broken?
look here:
https://github.com/pocoproject/poco/blob/master/CMakeLists.txt#L83
there is a code:
```
if(ENABLE_CRYPTO OR ENABLE_NETSSL OR ENABLE_JWT)
	find_package(OpenSSL REQUIRED)
else()
	find_package(OpenSSL)
endif()
```
So it seems that if there is openssl on build machine it will link against.
I see you are changing this behavior here, but I am having a trouble understanding whether it will actually correct it.

Shouldnt we patch out that cmake.list to not find_package(OpenSSL)  if the conanfile does not specify it?

Reply:
You're correct, it can pick up a system openssl.

The poco build system/recipe is broken in more ways.
I just noticed that `FORCE_OPENSSL` is not used anywhere in the poco build system.

Reply:
I think we are battling an issue internally for some time now, where new project linked into our binary causes coredump on deinit, 
we suspect that its because the system openssl has been taken into build phase from build box - and thats an ancient build box as you know :)

Reply:
It's not as bad as it seems.
It doesn't look like the openssl libraries are used anywhere except for setting the default options values of its dependencies.